app/components/task/archive-task.js

```diff
@@ -1,11 +1,20 @@
 import Component from '@ember/component';
 import { alias } from '@ember/object/computed';
+import { get } from '@ember/object';
 import EmberCan from 'ember-can';
 
 export default Component.extend({
   classNames: ['archive-task'],
 
   // auto-assigns 'task' property from component as ability 'model'
   ability: EmberCan.computed.ability('task'),
-  canArchive: alias('ability.canArchive')
+  canArchive: alias('ability.canArchive'),
+
+  actions: {
+    archiveTask() {
+      if (window.confirm(`Are you sure you want to archive this task? Archiving this task will remove it from your ${get(this, 'task.title')} task list.`)) {
```

Comment:
Missed this on the first pass, but you're doing `get(this, 'task.title')` instead of `get(this, 'task.taskList.title')`